core/domain/lib/src/usecase/user/user_settings_usecase.dart

```diff
@@ -0,0 +1,33 @@
+import 'package:domain/domain.dart';
+import 'package:domain/src/usecase/base/base_usecase.dart';
+
+class UserSettingsUsecase
+    extends BaseUseCase<BaseError, UserSettingsUsecaseParams, UserSettings> {
+  UserRepository _userRepository;
+
+  UserSettingsUsecase(this._userRepository);
+
+  @override
+  Future<Either<BaseError, UserSettings>> execute(
+      {required UserSettingsUsecaseParams params}) {
+    return _userRepository.getUserSettings(userId: params.userId);
+  }
+}
+
+class UserSettingsUsecaseParams extends Params {
+  int userId;
+
+  UserSettingsUsecaseParams(this.userId);
+
+  @override
+  Either<AppError, bool> verify() {
+    if (userId == 0) {
```

Comment:
use common function for this validation in other usecases

Reply:
Done